src/lib.rs

```diff
@@ -0,0 +1,182 @@
+use colored::Colorize;
+use std::{
+    fmt,
+    time::{Duration, Instant},
+};
+
+const RUNTIME_THRESHOLD: Duration = Duration::from_millis(5);
+
+pub struct FunctionBenchmark {
+    pub runtime: Option<Duration>,
+    memory_size: Option<u32>,
+    stack_size: Option<u32>,
+    start: Option<Instant>,
+}
+
+impl FunctionBenchmark {
+    /// Create a new `FunctionBenchmark`
+    pub fn new() -> Self {
+        FunctionBenchmark {
+            runtime: None,
+            memory_size: None,
+            stack_size: None,
+            start: None,
+        }
+    }
+
+    /// Start the benchmark
+    pub fn start(&mut self) {
+        self.start = Some(Instant::now());
+    }
+
+    /// Stop the benchmark
```

Comment:
This comments are superfluous. We should avoid comments unless they're absolutely necessary or being used to generate documentation.

Reply:
Perfect, I'll remove them